llvm/lib/Target/RISCV/RISCVISelLowering.cpp

```diff
@@ -1884,6 +1884,21 @@ bool RISCVTargetLowering::isTruncateFree(EVT SrcVT, EVT DstVT) const {
   return (SrcBits == 64 && DestBits == 32);
 }
 
+bool RISCVTargetLowering::isTruncateFree(SDValue Val, EVT VT2) const {
+  EVT SrcVT = Val.getValueType();
+  // free truncate from vnsrl and vnsra
+  if (Subtarget.hasStdExtV() &&
```

Comment:
I'm late to this, but shouldn't this use hasVInstructions(). It's excluding Zve*.

Reply:
Fixed by 3de76e4f57bdba612ddca7ac53b2f5787929391a